LIS3DH.device.lib.nut

```diff
@@ -0,0 +1,486 @@
+// Registers
+const LIS3DH_TEMP_CFG_REG  = 0x1F;
+const LIS3DH_CTRL_REG1     = 0x20;
+const LIS3DH_CTRL_REG2     = 0x21;
+const LIS3DH_CTRL_REG3     = 0x22;
+const LIS3DH_CTRL_REG4     = 0x23;
+const LIS3DH_CTRL_REG5     = 0x24;
+const LIS3DH_CTRL_REG6     = 0x25;
+const LIS3DH_OUT_X_L_INCR  = 0xA8;
+const LIS3DH_OUT_X_L       = 0x28;
+const LIS3DH_OUT_X_H       = 0x29;
+const LIS3DH_OUT_Y_L       = 0x2A;
+const LIS3DH_OUT_Y_H       = 0x2B;
+const LIS3DH_OUT_Z_L       = 0x2C;
+const LIS3DH_OUT_Z_H       = 0x2D;
+const LIS3DH_FIFO_CTRL_REG = 0x2E;
+const LIS3DH_FIFO_SRC_REG  = 0x2F;
+const LIS3DH_INT1_CFG      = 0x30;
+const LIS3DH_INT1_SRC      = 0x31;
+const LIS3DH_INT1_THS      = 0x32;
+const LIS3DH_INT1_DURATION = 0x33;
+const LIS3DH_CLICK_CFG     = 0x38;
+const LIS3DH_CLICK_SRC     = 0x39;
+const LIS3DH_CLICK_THS     = 0x3A;
+const LIS3DH_TIME_LIMIT    = 0x3B;
+const LIS3DH_TIME_LATENCY  = 0x3C;
+const LIS3DH_TIME_WINDOW   = 0x3D;
+const LIS3DH_WHO_AM_I      = 0x0F;
+    
+
+// Bitfield values
+const LIS3DH_X_LOW         = 0x01;
+const LIS3DH_X_HIGH        = 0x02;
+const LIS3DH_Y_LOW         = 0x04;
+const LIS3DH_Y_HIGH        = 0x08;
+const LIS3DH_Z_LOW         = 0x10;
+const LIS3DH_Z_HIGH        = 0x20;
+const LIS3DH_SIX_D         = 0x40;
+const LIS3DH_AOI           = 0x80;
+
+// High Pass Filter values
+const LIS3DH_HPF_DISABLED               = 0x00;
+const LIS3DH_HPF_AOI_INT1               = 0x01;
+const LIS3DH_HPF_AOI_INT2               = 0x02;
+const LIS3DH_HPF_CLICK                  = 0x04;
+const LIS3DH_HPF_FDS                    = 0x08;
+
+const LIS3DH_HPF_CUTOFF1                = 0x00;
+const LIS3DH_HPF_CUTOFF2                = 0x10;
+const LIS3DH_HPF_CUTOFF3                = 0x20;
+const LIS3DH_HPF_CUTOFF4                = 0x30;
+
+const LIS3DH_HPF_DEFAULT_MODE           = 0x00;
+const LIS3DH_HPF_REFERENCE_SIGNAL       = 0x40;
+const LIS3DH_HPF_NORMAL_MODE            = 0x80;
+const LIS3DH_HPF_AUTORESET_ON_INTERRUPT = 0xC0;
+
+const LIS3DH_FIFO_BYPASS_MODE           = 0x00;
+const LIS3DH_FIFO_FIFO_MODE             = 0x40;
+const LIS3DH_FIFO_STREAM_MODE           = 0x80;
+const LIS3DH_FIFO_STREAM_TO_FIFO_MODE   = 0xC0;
+
+// Click Detection values
+const LIS3DH_SINGLE_CLICK  = 0x15;
+const LIS3DH_DOUBLE_CLICK  = 0x2A;
+
+const LIS3DH_MODE_NORMAL = 0x00;
+const LIS3DH_MODE_LOW_POWER = 0x01;
+const LIS3DH_MODE_HIGH_RESOLUTION = 0x02;
+
+const LIS3DH_ADC1 = 0x01;
+const LIS3DH_ADC2 = 0x02;
+const LIS3DH_ADC3 = 0x03;
+
+class LIS3DH {
+    static VERSION = "2.0.0";
+
+    // I2C information
+    _i2c = null;
+    _addr = null;
+    _mode = LIS3DH_MODE_NORMAL;
+
+    // The full-scale range (+/- _range G)
+    _range = null;
+
+    constructor(i2c, addr = 0x30) {
+        _i2c = i2c;
+        _addr = addr;
+
+        // Read the range + set _range property
+        getRange();
+    }
+    
+    // set default values for registers, read the current range and set _range
+    // (resets to state when first powered on)
+    function init() {
```

Comment:
So the method doesn't have to be called on cold start correct? If so, why don't we call it `reset` or something? Wouldn't it be a more accurate description?